sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyRequestParameters.cs

```diff
@@ -77,13 +77,19 @@ internal KeyRequestParameters(KeyType type, KeyCreateOptions options = default)
         internal KeyRequestParameters(EcKeyCreateOptions ecKey)
             : this(ecKey.KeyType, ecKey)
         {
-            Curve = ecKey.Curve;
+            if(ecKey.Curve != default)
+            {
+                Curve = ecKey.Curve;
+            }
         }
 
         internal KeyRequestParameters(RsaKeyCreateOptions rsaKey)
             : this(rsaKey.KeyType, rsaKey)
         {
-            KeySize = rsaKey.KeySize;
+            if(rsaKey.KeySize != default)
+            {
+                KeySize = rsaKey.KeySize;
+            }
```

Comment:
Same comment as above.  Although 0 isn't a valid value for KeySize we should make KeySize nullable on RsaKeyCreateOptions so if the user does explicitly set a value we don't ignore it (in this case we would never send 0)

Reply:
yeah.. I'll change it.